app/src/main/java/com/example/mvvmcleanarchitecture/core/di/AppModule.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.mvvmcleanarchitecture.core.di
+
+import android.content.Context
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import org.koin.android.ext.koin.androidContext
+import org.koin.dsl.module
+
+val appModule = module {
+    factory { provideLinearLayoutManager(androidContext()) }
+
+    factory { provideGridLayoutManager(androidContext()) }
+
+    factory { provideDividerItemDecoration(androidContext()) }
+}
+
+private fun provideLinearLayoutManager(androidContext: Context): LinearLayoutManager {
+    return LinearLayoutManager(androidContext)
+}
+
+private fun provideGridLayoutManager(androidContext: Context): GridLayoutManager {
```

Comment:
Czy jeżeli mamy możliwość definiowania np. w xmlu takich atrybutów pod recyclerView to powinniśmy wstrzykiwać to przez DI czy uzupełniać takie atrybuty w xmlu ? Zakładając że będzie on reużywalny hardkodoujemy w nim zachowanie np: app:layoutManager="androidx.recyclerview.widget.GridLayoutManager" ?

Reply:
W zasadzie odpowiedziałem na to pytanie wyżej ☝️